common/persistence/sql/sqlShardPersistence.go

```diff
@@ -0,0 +1,359 @@
+			row, err := shardInfoToShardsRow(*request.ShardInfo)
+			if err != nil {
+				return &workflow.InternalServiceError{
+					Message: fmt.Sprintf("CreateShard operation failed. Error: %v", err),
+				}
+			}
+
+			if _, err := m.db.NamedExec(createShardSQLQuery, row); err != nil {
+				return nil
+			}
+		default:
+			return &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateShard operation failed. Failed to verify whether a shard of ID %v already existed. Error: %v", request.ShardInfo.ShardID, err),
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *sqlShardManager) GetShard(request *persistence.GetShardRequest) (*persistence.GetShardResponse, error) {
+	stmt, err := m.db.PrepareNamed(getShardSQLQuery)
+	if err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to prepare statement. ShardId: %v. Error: %v", request.ShardID),
+		}
+	}
+
+	var row shardsRow
+	err = stmt.Get(&row, struct {
+		ShardID int `db:"shard_id"`
+	}{request.ShardID})
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("GetShard operation failed. Shard with ID %v not found. Error: %v", request.ShardID, err),
+			}
+		}
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to get record. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
+	clusterTransferAckLevel := make(map[string]int64)
+	if err := gobDeserialize(row.ClusterTransferAckLevel, &clusterTransferAckLevel); err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to deserialize ShardInfo.ClusterTransferAckLevel. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
+	clusterTimerAckLevel := make(map[string]time.Time)
+	if err := gobDeserialize(row.ClusterTimerAckLevel, &clusterTimerAckLevel); err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to deserialize ShardInfo.ClusterTimerAckLevel. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
+	return &persistence.GetShardResponse{ShardInfo: &persistence.ShardInfo{
+		ShardID:                   int(row.ShardID),
+		Owner:                     row.Owner,
+		RangeID:                   row.RangeID,
+		StolenSinceRenew:          int(row.StolenSinceRenew),
+		UpdatedAt:                 row.UpdatedAt,
+		ReplicationAckLevel:       row.ReplicationAckLevel,
+		TransferAckLevel:          row.TransferAckLevel,
+		TimerAckLevel:             row.TimerAckLevel,
+		ClusterTransferAckLevel:   clusterTransferAckLevel,
+		ClusterTimerAckLevel:      clusterTimerAckLevel,
+		DomainNotificationVersion: row.DomainNotificationVersion,
+	}}, nil
+}
+
+func (m *sqlShardManager) UpdateShard(request *persistence.UpdateShardRequest) error {
+	_, err := m.GetShard(&persistence.GetShardRequest{request.ShardInfo.ShardID})
```

Comment:
What if we don't do get? update will fail anyways, right?
Also since we only have one update statement, we probably don't need the lock and transaction.